lib/improver/cli/nbhood_iterate_with_mask.py

```diff
@@ -133,26 +134,105 @@ def main(argv=None):
     cube = load_cube(args.input_filepath)
     mask_cube = load_cube(args.input_mask_filepath)
 
-    if args.radius:
-        radius_or_radii = args.radius
-        lead_times = None
-    elif args.radii_by_lead_time:
-        radius_or_radii = args.radii_by_lead_time[0].split(",")
-        lead_times = args.radii_by_lead_time[1].split(",")
+    weights = load_cube(args.weights_for_collapsing_dim) if \
+        args.collapse_dimension else None
 
-    result = ApplyNeighbourhoodProcessingWithAMask(
-        args.coord_for_masking, radius_or_radii, lead_times=lead_times,
-        sum_or_fraction=args.sum_or_fraction,
-        re_mask=args.re_mask).process(cube, mask_cube)
+    result, intermediate_cube = process(
+        cube, mask_cube, weights, args.coord_for_masking, args.radius,
+        args.radii_by_lead_time, args.sum_or_fraction, args.re_mask,
+        args.collapse_dimension)
 
+    save_netcdf(result, args.output_filepath)
     if args.intermediate_filepath is not None:
-        save_netcdf(result, args.intermediate_filepath)
+        save_netcdf(intermediate_cube, args.intermediate_filepath)
+
+
+def process(cube, mask_cube, weights, coord_for_masking, radius,
+            radii_by_lead_time, sum_or_fraction="fraction", re_mask=False,
+            collapse_dimension=False):
+    """Runs neighbourhooding processing iterating over a coordinate by mask.
+
+    Apply the requested neighbourhood method via the
+    ApplyNeighbourhoodProcessingWithMask plugin to a file with one diagnostic
+    dataset in combination with a cube containing one or more masks.
+    The mask dataset may have an extra dimension compared to the input
+    diagnostic. In this case, the user specifies the name of the extra
+    coordinate and this coordinate is iterated over so each mask is applied
+    to separate slices over the input cube. These intermediate masked datasets
+    are then concatenated, resulting in a dataset that has been processed
+    using multiple masks and has gained an extra dimension from the masking.
+    There is also an option to re-mask the output dataset, so that after
+    neighbourhood processing non-zero values are only present for unmasked
+    grid points.
+    There is an alternative option of collapsing the dimension that we gain
+    using this processing using a weighted average.
+
+    Args:
+        cube (iris.cube.Cube):
+            Cube to be processed.
+        mask_cube (iris.cube.Cube):
+            Cube to act as a mask.
+        weights (iris.cube.Cube):
+            Cube containing the weights which are used for collapsing the
+            dimension gained through masking.
+        coord_for_masking (str):
+            String matching the name of the coordinate that will be used
+            for masking.
+        radius (float):
+            The radii in metres of the neighbourhood to apply.
```

Comment:
Seems strange to have 'radii' in the description when the variable is a single 'radius'